core/src/com/unciv/ui/pickerscreens/FoundReligionPickerScreen.kt

```diff
@@ -0,0 +1,102 @@
+
+import com.badlogic.gdx.scenes.scene2d.ui.*
+import com.badlogic.gdx.utils.Align
+import com.unciv.UncivGame
+import com.unciv.logic.GameInfo
+import com.unciv.logic.civilization.CivilizationInfo
+import com.unciv.models.UncivSound
+import com.unciv.models.ruleset.Belief
+import com.unciv.models.translations.tr
+import com.unciv.ui.utils.*
+import kotlin.math.max
+
+class FoundReligionPickerScreen (
+    private val choosingCiv: CivilizationInfo,
+    private val gameInfo: GameInfo
+): PickerScreen() {
+
+    // Roughly follows the layout of the original (although I suck at UI designing, so please improve this)
+    private val topReligionIcons = Table() // Top of the layout, contains icons for religions
+    private val leftChosenBeliefs = Table() // Left middle part, contains buttons to select the types of beliefs to choose
+    private val rightBeliefsToChoose: ScrollPane // Right middle part, contains the beliefs to choose
+    
+    private val middlePanes: SplitPane
+    private val newPanes: SplitPane
+ 
+    private var previouslySelectedIcon: Button? = null
+    private var iconName: String? = null
+    private var religionName: String? = null
+    private var chosenFounderBelief: Belief? = null
+    private var chosenFollowerBelief: Belief? = null
+
+    init {
+        closeButton.isVisible = true
+        setDefaultCloseAction()
+        
+        stage.clear()
+        
+        setupReligionIcons()
+                
+        rightBeliefsToChoose = ScrollPane(Table())
+        
+        descriptionLabel.center(bottomTable)
+        
+        middlePanes = SplitPane(leftChosenBeliefs, rightBeliefsToChoose, false, skin)
+        middlePanes.splitAmount = 0.5f
+        
+        newPanes = SplitPane(topReligionIcons, middlePanes, true, skin)
+        newPanes.splitAmount = 0.15f
+        
+        splitPane = SplitPane(newPanes, bottomTable, true, skin)
+        splitPane.splitAmount = screenSplit
+        splitPane.setFillParent(true)
+        stage.addActor(splitPane)
+        
+        rightSideButton.label = "Choose a religion".toLabel()
+        rightSideButton.onClick(UncivSound.Choir) {
+            choosingCiv.religionManager.foundReligion(iconName!!, religionName!!, "", "", /**chosenFollowerBelief!!.name, chosenFounderBelief!!.name*/)            
+            UncivGame.Current.setWorldScreen()
+        }
+    }
+    
+    private fun checkAndEnableRightSideButton() {
+        if (religionName == null) return
+        // check if founder belief chosen
+        // check if follower belief chosen
+        rightSideButton.enable()
+    }
+
+    private fun setupReligionIcons() {
+        topReligionIcons.clear()
+        
+        val descriptionLabel = Label("Choose an Icon and name for your Religion".tr(), skin)
+        
+        val iconsTable = Table()
+        iconsTable.align(Align.center)
+        for (religionName in listOf("Christianity".tr(), "Islam".tr(), "Taoism".tr(), "Hinduism".tr(), "Buddhism".tr())) {
```

Comment:
Hm. This will do for now, but I'm not sure where would be the right place to p these.
modOptions perhaps?

Reply:
I could create a "Religions.json" file which is just a list of the names of all religions, but that does kind of feel like overkill, though it would be more modular than this. ModOptions doesn't feel right either, that should (in my opinion) only include data about the mod, not game-related data.

Reply:
Another option is to cheat and use the religion images as the source of names.
Like ImageGetter.textureRegionDrawables.keys.filter {it.startsWith("ReligionIcons")}

Reply:
That however will limit the mods, since the images will ALWAYS contain the religion icons that come in the base game, you'd only be able to add to them, not remove